medusa/providers/torrent/html/archetorrent.py

```diff
@@ -0,0 +1,215 @@
+
+                results += self.parse(response.text, mode, keywords=search_string)
+
+        return results
+
+    def parse(self, data, mode, **kwargs):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS.
+
+        :return: A list of items found
+        """
+        # Units
+        units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB']
+
+        items = []
+
+        keywords = kwargs.pop('keywords', None)
+
+        with BS4Parser(data, 'html5lib') as html:
+            torrent_table = html.find(class_='ttable_headinner')
+            torrent_rows = torrent_table('tr') if torrent_table else []
+
+            # Continue only if at least one release is found
+            if len(torrent_rows) < 2:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            # Catégorie, Release, Date, DL, Size, C, S, L
+            labels = [label.get_text(strip=True) for label in torrent_rows[0]('th')]
+
+            for torrent in torrent_rows[1:]:
+                cells = torrent('td')
+                if len(cells) < len(labels):
+                    continue
+
+                try:
+                    torrent_id = re.search('id=([0-9]+)', cells[labels.index('Nom')].find('a')['href']).group(1)
+                    title = cells[labels.index('Nom')].get_text(strip=True)
+                    download_url = urljoin(self.urls['download'], '?id={0}&name={1}'.format(torrent_id, title))
+                    if not all([title, download_url]):
+                        continue
+
+                    # Chop off tracker/channel prefix or we cannot parse the result!
+                    if mode != 'RSS' and keywords:
+                        show_name_first_word = re.search(r'^[^ .]+', keywords).group()
+                        if not title.startswith(show_name_first_word):
+                            title = re.sub(r'.*(' + show_name_first_word + '.*)', r'\1', title)
+
+                    seeders = try_int(cells[labels.index('S')].get_text(strip=True))
+                    leechers = try_int(cells[labels.index('L')].get_text(strip=True))
+
+                    # Filter unseeded torrent
+                    if seeders < min(self.minseed, 1):
+                        if mode != 'RSS':
+                            log.debug("Discarding torrent because it doesn't meet the"
+                                      " minimum seeders: {0}. Seeders: {1}",
+                                      title, seeders)
+                        continue
+
+                    size_index = labels.index('Size') if 'Size' in labels else labels.index('Taille')
+                    torrent_size = cells[size_index].get_text()
+                    size = convert_size(torrent_size, units=units) or -1
+
+                    pubdate_raw = BS4Parser(torrent('a')[1]['onmouseover'][16:-2],
```

Comment:
Why is it again creating a BS4parser here?

Reply:
This one is tricky and I spent time searching for a solution. This provider's way of displaying the pubdate is awful. Is is stored in a onMouseOver attribute, within a "return" statement:

`<a href="torrents-details.php?id=82486&amp;hit=1" onMouseover="return overlib('<table width=100% class=ttable_col1 cellspacing=0 cellpadding=5 align=center><tr><td class=ttable_head colspan=2 align=center>Game.Over.Le.Regne.Des.Jeux.Videos.2013.FRENCH.HDTV.720P.AVC.AC3-...</td></tr><tr valign=top><td align=center><img border=0 width=100 src=https://archetorrent.com/uploads/images/824860.png></td><td><div align=left><b>Poster le: </b>14-11-2017<br /><b>Heure: </b>19:00<br><b>Taille: </b>598.33 MB<br /><b>Terminer: </b>5<br /></div><div align=left><b>Vue: </b>14<br /><b>Langue: </b>Français<br /><b>Hits: </b>7<br /><b>Seeders: </b><font color=green>4</font><br /><b>Leechers: </b><font color=red>0</font> </div></td></tr></table>')"; onMouseout="return nd()"><b>Game.Over.Le.Regne.Des.Jeux.Videos.2013.FRENCH.HDTV.720P.AVC.AC3-...</b> <img src='images/silver.gif' border='0' alt='' /> <img src='images/seedbox.png' border='0' title='Torrent seedbox' /></a>`

We want to capture 14-11-2017 which has no surrounding tag.

First, I get the link using torrent('a')[1]['onmouseover'], then I have to remove the "return overlib('" at the beginning and "')" at the end. At this point, this is just a unicode string, not a BS4 object anymore. To get the actual date, I reparse it and get the appropriate content.

I wish there was a cleaner way to do it, let me know if you can this of something.

![a](https://user-images.githubusercontent.com/959078/32915202-0e68aaee-cae6-11e7-8f89-26ead7e2ca3b.png)

Reply:
Instead of using `BS4Parser` again, you could get the content of `onMouseover` with get_text() and use a regex (e.g. `<b>Poster le: </b>(\d{2}-\d{2}-\d{4})`) to get the date. It should be definetely faster.